### ISSUE TITLE
[Question]: Using flair for creating tags or keywords that represents a given text

### ISSUE THREAD

**Issue by B0rner** [flairNLP/flair#3443]
### Question

Hey,
I wonder if it is possible to use one of the flair model (for example provided over huggingface) in such a way that it generates keywords from a text that is passed to the data model. Preferably in German.

I tried `ner-english-ontonotes-large` which works very well on German text for specific entities.

But what I need is model, that returns not specific entities, like the name of the person or the name of the city. I need a model, that returns a couple of keywords, that describes the context of the input text, like `input:  "I like to have pizza for breakfast." `output: "breakfast", "pizza", "favorite meal"` ..which should also work for longer texts.

There is no model in the flair-oniverse, that fits exactly that usecase, right?

But is it possible, to train or fine-tune a flair-model for that? A have a huge set of data, that provides a text and related keywords, which would be ideal for finetuning a model. Is there a way (and a tutorial), how to feed the trainig-data to make the model fit for the usecase?

**Comment by fkdosilovic** [comment]
> ... I need a model, that returns a couple of keywords, that describes the context of the input text, like input:  "I like to have pizza for breakfast." output: "breakfast", "pizza", "favorite meal"` ..which should also work for longer texts.

What you described is a keyword extraction task. Your best bet is to look at the [KeyBERT library](https://maartengr.github.io/KeyBERT/index.html).

> I wonder if it is possible to use one of the flair model (for example provided over huggingface) in such a way that it generates keywords from a text that is passed to the data model. Preferably in German.

If you want to "generate" your output, maybe look into [KeyLLM](https://maartengr.github.io/KeyBERT/api/keyllm.html) from the KeyBERT library.

> There is no model in the flair-oniverse, that fits exactly that usecase, right?

From the [API docs](https://flairnlp.github.io/flair/master/api/flair.models.html) of flair, there doesn't seem to be anything like with KeyBERT.

> But is it possible, to train or fine-tune a flair-model for that? A have a huge set of data, that provides a text and related keywords, which would be ideal for finetuning a model. Is there a way (and a tutorial), how to feed the trainig-data to make the model fit for the usecase?

Maybe you can train a multi-label classifier, where inputs are documents and labels are keywords that describe best describe the document.

**Comment by B0rner** [comment]
Hey,

thanks four your answers.

>What you described is a keyword extraction task. Your best bet is to look at the [KeyBERT library](https://maartengr.github.io/KeyBERT/index.html).

Sound interesting. But it only extracts exiting words from the text. I was looking for a model, that generates abstract keywords, like `"favorite meal"` from the text `"I like to have pizza for breakfast."`

>Maybe you can train a multi-label classifier, where inputs are documents and labels are keywords that describe best describe the document.

Yes, this is, what I did. I used BART and inputs a set of documents and labels. I think, that works okay, but i like to take a sledgehammer to crack a nut.
